gateway/gateway-runtime/policy-engine/internal/kernel/stream_compression_test.go

```
@@ -0,0 +1,224 @@
+
+// singlePassGunzip decodes exactly like a normal HTTP client: one reader over
+// the whole payload, stopping at the end of the first gzip member. This is what
+// httpx/urllib3, curl and Go's own transport do — so it is the invariant that
+// actually matters, not whether the bytes are on the wire somewhere.
+func singlePassGunzip(t *testing.T, wire []byte) []byte {
+	t.Helper()
+	zr, err := gzip.NewReader(bytes.NewReader(wire))
+	if err != nil {
+		t.Fatalf("gzip.NewReader: %v", err)
+	}
+	defer zr.Close()
```

golangci-lint (2.12.2)

[error] Error return value of `zr.Close` is not checked

(errcheck)